### PR TITLE
add a test for popupMenuButton

### LINKED ISSUE #80869: PopupMenuButton building list on the incorrect place
When i use the PopupMenuButton to create a list inside a scrollview, the menu opens on the same location of the screen  instead of having a constant behavior related to the button.

The error starts happening in newer flutter versions (the last week ones)

The test code:

```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

String _quebra = "";
List<String> status = ["ONE", "TWO", "OPEN", "CLOSED"];

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: SingleChildScrollView(
              child: Center(
                child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            placeholder(),
            placeholder(),
            Text(
                'You have pushed the button this many times:',
            ),
            Text(
                '$_counter',
                style: Theme.of(context).textTheme.headline4,
            ),
            SizedBox(
                height: 40,
            ),
            Text(
                _quebra,
                style: TextStyle(fontWeight: FontWeight.bold, fontSize: 20),
            ),
            SizedBox(
                height: 40,
            ),
            PopupMenuButton(
                padding: const EdgeInsets.only(top: 5.0),
                icon: Icon(
                  Icons.keyboard_arrow_down,
                  color: Colors.red,
                ),
                offset: Offset(80, 40),
                onSelected: popupMenuSelected,
                itemBuilder: (BuildContext context) => <PopupMenuEntry<String>>[
                  buildPopItem("Todos"),
                  buildPopItem(status[0]),
                  buildPopItem(status[1]),
                  buildPopItem(status[2]),
                ],
            ),
            placeholder(),
            placeholder(),
            placeholder(),
          ],
        ),
              ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ),
    );
  }

  void popupMenuSelected(String valueSelected) async {
    setState(() {
      _quebra = valueSelected;
    });
  }
}

PopupMenuItem<String> buildPopItem(String texto, {bool enabled = true}) {
  return PopupMenuItem<String>(
    value: texto,
    child: Text(texto),
    enabled: enabled,
  );
}

Widget placeholder() {
  return Column(
    children: [
      SizedBox(
        height: 70,
      ),
      Text("placeholder",style: TextStyle(fontWeight: FontWeight.bold, fontSize: 40)),
    ],
  );
}
```

my doctor:

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, 2.2.0-11.0.pre.191, on Microsoft Windows [versÃ£o 10.0.19042.928],
    locale pt-BR)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.    
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code (version 1.55.2)
[√] Connected device (3 available)
```

### PR DESCRIPTION
**Pull request by xu-baolin**
Fixes #80869

### CODE REVIEW

**Comment by xu-baolin** [comment]
The root cause of the original change is that the scroll event did not trigger the position update.
https://github.com/xu-baolin/flutter/blob/b7436456832a11403a58e5f2f4245f622da9515a/packages/flutter/lib/src/material/popup_menu.dart#L1197

File: packages/flutter/test/material/popup_menu_test.dart [hunk 1]
```diff
@@ -2209,6 +2209,70 @@ void main() {
     await tester.pumpAndSettle();
     expect(find.text('foo'), findsOneWidget);
   });
+
+  // Regressing test for https://github.com/flutter/flutter/issues/80869
```
Comment (shihaohong) [inline packages/flutter/test/material/popup_menu_test.dart:2213]:
```suggestion
  // Regression test for https://github.com/flutter/flutter/issues/80869
```

File: packages/flutter/test/material/popup_menu_test.dart [hunk 2]
```diff
@@ -2209,6 +2209,70 @@ void main() {
...
+                    child: const ColoredBox(
+                      color: Colors.pink,
+                    ),
+                  ),
+                  itemBuilder: (BuildContext context) => <PopupMenuEntry<int>>[
+                    const PopupMenuItem<int>(child: Text('-1-'), value: 1,),
+                    const PopupMenuItem<int>(child: Text('-2-'), value: 2,),
+                  ],
+                ),
+                const SizedBox(height: 600,),
```
Comment (shihaohong) [inline packages/flutter/test/material/popup_menu_test.dart:2238]: nits:
```suggestion
                const SizedBox(height: 100),
                PopupMenuButton<int>(
                  child: SizedBox(
                    key: buttonKey,
                    height: 10.0,
                    width: 10.0,
                    child: const ColoredBox(
                      color: Colors.pink,
                    ),
                  ),
                  itemBuilder: (BuildContext context) => <PopupMenuEntry<int>>[
                    const PopupMenuItem<int>(child: Text('-1-'), value: 1,),
                    const PopupMenuItem<int>(child: Text('-2-'), value: 2,),
                  ],
                ),
                const SizedBox(height: 600),
```

**Review by shihaohong — APPROVED** [review]
LGTM with nits

**Pull request merged**